test/test.js

```diff
@@ -0,0 +1,295 @@
+      if (found) {
+        return estraverse.VisitorOption.Skip;
+      }
+      if (node.type == type) {
+        if (skipped === skip) {
+          found = node;
+          return estraverse.VisitorOption.Skip;
+        }
+        skipped++;
+      }
+    }
+  });
+
+  if (!found) {
+    throw new Error('did not find AwaitExpression (skipped ' + skipped + '/' + skip + ')');
+  }
+
+  return found;
+}
+
+function extendOptions(pluginOptions, acornOptions) {
+  return xtend({
+    sourceType: 'module',
+    ecmaVersion: 7,
+    locations: true,
+    ranges: true,
+    plugins: {asyncawait: pluginOptions || pluginOptions !== false}
+  }, acornOptions);
+}
+
+function parse(code, pluginOptions, acornOptions) {
+  if (Array.isArray(code)) {
+    code = code.join('\n');
+  }
+  var options = extendOptions(pluginOptions, acornOptions);
+  return acorn.parse(code, options);
+}
+
+describe('async', () => {
+  describe ('function declaration', () => {
+    var node;
+
+    beforeEach(() => {
+      node = find(
+        'FunctionDeclaration',
+        parse([
+          'async function foo() {',
+          '  x = await bar()',
+          '}'
+        ])
+      );
+    });
+
+    it('marks the node as async', () =>
+      assert(node.async)
+    );
+
+    it('finds correct start position', () =>
+      assert.strictEqual(node.start, 0)
+    );
+
+    it('finds correct end position', () =>
+      assert.strictEqual(node.end, 42)
+    );
+
+    it('finds correct start line/column', () =>
+      assert.deepEqual(node.loc.start, {
+        line: 1,
+        column: 0
+      })
+    );
+
+    it('finds correct end line/column', () =>
+      assert.deepEqual(node.loc.end, {
+        line: 3,
+        column: 1
+      })
+    );
+  });
+
+  describe ('function expression', () => {
+    var node, code;
+
+    beforeEach(() => {
+      code = [
+        'foo = async function () {',
+        '  x = await bar()',
+        '}'
+      ];
+      node = find(
+        'FunctionExpression',
+        parse(code)
+      );
+    });
+
+    it('marks the node as async', () =>
+      assert(node.async)
+    );
+
+    it('finds correct start position', () =>
+      assert.strictEqual(node.start, 6)
+    );
+
+    it('finds correct end position', () =>
+      assert.strictEqual(node.end, code.join('\n').length)
+    );
+
+    it('finds correct start line/column', () =>
+      assert.deepEqual(node.loc.start, {
+        line: 1,
+        column: 6
+      })
+    );
+
+    it('finds correct end line/column', () =>
+      assert.deepEqual(node.loc.end, {
+        line: 3,
+        column: 1
+      })
+    );
+  });
+
+  describe ('enhanced object literal', () => {
+    var node, code;
+
+    beforeEach(() => {
+      code = [
+        'var x = {',
+        '  async foo() {}',
+        '};'
+      ];
+      node = find(
+        // TODO: Is it really supposed to mark the Property async? Why not the FunctionExpression?
+        'Property',
+        parse(code)
```

Comment:
I find it surprising the `async` modifier gets placed on the `Property` and not the `FunctionExpression`. Are you sure that is right? (I'm guessing you know what you are doing :smile:).